doc/dev/error-message.md

```diff
@@ -0,0 +1,20 @@
+# Error message
+
+Elektra has a lot of developers and hence the error messages vary in how they are written.
+This guideline should unify the message format so that users will get consistent error messages.
+
+- All error messages start with a capital letter. The only exception are quoted messages at the start such as "'xxxx-xxxx' is an invalid syntax".
+  If you reference a method name such as `kdbSet()` in the first word, surround it with single quotes.
+  Try to avoid variable strings at the start of the sentence ("%s"-strings) if possible because you cannot guarantee capital casing.
+- **Always** provide the errno if possible. If possible use `strerror_r` for writing out the message because it is thread safe.
+- If you can provide an errno, write `"Reason: %s", strerror(errno)` at the end of the error message. Also end the preceding sentence with a dot.
+  Example: `"XY failed. Reason: %s", strerror(errno)`. Alternatively you can also embed it into a standard text: `"XY failed because of %s", strerror(errno)`.
+- The last sentence (or single sentence of your message if you just provide one) must not end with a dot. This should encourage users to continue reading
```

Comment:
I do not think I like that style very much. In my opinion, sentences should always end with a dot.

Reply:
I agree but honestly I rarely have seen error messages in applications which end with a dot. Also it is kind of counterintuitive to write dots at the end of error messages.